beater/handler.go

```diff
@@ -0,0 +1,172 @@
+package beater
+
+import (
+	"compress/gzip"
+	"compress/zlib"
+	"crypto/subtle"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/elastic/apm-server/processor"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+var (
+	errInvalidToken    = errors.New("Invalid token")
+	errPOSTRequestOnly = errors.New("Only post requests are supported")
+)
+
+type handler func(w http.ResponseWriter, r *http.Request)
+
+func createHandler(p processor.Processor, config Config, report reporter) handler {
```

Comment:
Generally I like the idea of moving the handler part to its own file. When you move the logic, you should definitly also move the tests. This could be the tricky part since it is quite strictly coupled. 
So from my side +1 on seperating the logic, but not without the tests. I'd see this as a basis to see where we still have a lot of coupling, mixing and holding states instead of passing in values, which makes decoupling and testing harder.

Reply:
Will do if we agree on moving forward with this.